Add default file format when Quarto format is `typst`

elipousson opened the pull request:
Happy to open a GitHub issue for discussion but, since this is such a small PR, I thought I'd go ahead and give this a try.

I also added the DarkFlagshipTerrastruct to the list of supported themes (it appeared to be missing but I can reverse that change if I missed something).

rcannood commented on _extensions/d2/d2.lua:
```diff
@@ -121,13 +129,17 @@ local function render_graph(globalOptions)
       -- Set default filename
       if options.filename == nil then
-        options.filename = "diagram-" .. counter
+        options.filename = "diagram-" .. counter .. math.random (1, 1000)
```
Why add a random number to this filename?

elipousson replied:
I had an issue with the extension *not* overwriting existing files that this seemed to resolve. I’ll try to put together a reprex (it may be specifically related to using knit_child?) to see if this random number assignment is a real fix or if there may be a better way to fix the issue.

rcannood reviewed (commented):
Hey Eli!

Thanks for creating this PR.

Would you be able to add an entry to the CHANGELOG.md, describing each of the changes made in this PR?

In addition, I'm wondering why you add a random number to the filename, as this might lead to many duplicated files if the document is rerendered a couple of times.

Thanks!

elipousson commented:
Thanks for the review!

I’ll go back to the file naming code and make sure to document the necessity of the change.

The change is detecting the d2 attribute class also introduce a small bug that I haven’t quite fixed yet.

I’ll get that sorted along with the CHANGELOG in the next week or two. Thanks for a cool extension!

elipousson commented:
I think I've almost got it right but I need to work out one more bug. I'll flag when it is ready for review again.

elipousson commented:
Dang, I think I've got some good improvementsin this PR—support for external d2 files, more consistent handling of output folder paths, and those improvements for tu[st output formats—but I'm 100% stuck on getting the raw embeds working again.

If you have a chance to take a look, @rcannood, I'd love to know what I'm doing wrong.

elipousson commented:
Finally figured it out: I had an error in how I was writing the control flow in Lua but otherwise it all works as expected. Ready for a second review, I think!

rcannood commented:
Happy to see you managed to find the issue! I intended to take a look at it, but didn't manage to find the time.

Could you merge the main branch into yours and move your examples to `tests/`? Currently the CI will only render the qmds, but it's better than nothing :)

elipousson commented:
Added some basic examples to tests (along with a simple d2 file for testing).

I also added some partial support for code folding per the discussion on #6. Code folding will work in HTML output but there are some unresolved formatting issues with d2 code blocks (the gray background doesn't show in the HTML format).

rcannood commented:
LGTM!

Thanks @elipousson for making significant changes to the codebase!

Would you like to become a collaborator in this repository?

elipousson commented:
@rcannood I'd be happy to! Time permitting, I'd love to figure out how to implement Quarto-style `#|` block options using https://github.com/coatless-quarto/codecelloptions if you're up for having it added. I also just spotted a typo (`mt` in the pandoc.insert.mediabag call instead of `mimetype`) which I can fix directly. I appreciate the invitation and the opportunity to contribute to such a handy extension.
